### PR TITLE
Update RELEASE_NOTE_DRAFT.md.

### PR DESCRIPTION
**Pull request by donraab**
Signed-off-by: Donald Raab <Donald.Raab@bnymellon.com>

### CODE REVIEW

**Comment by nikhilnanivadekar** [comment]
I would like to merge this PR as the last PR for 9.0.0. 
Also make a few changes for rearranging.

**Comment by nikhilnanivadekar** [comment]
I am using this PR as an opportunity to clean up the release notes. Sorry that I am requesting changes for work done in an older commit.

**Comment by donraab** [comment]
Ok will update and re-submit.

File: RELEASE_NOTE_DRAFT.md [hunk 1]
```diff
@@ -6,36 +6,34 @@ This is the 9.0.0 major release.
...
---------------------------
-
 * For Java 9 add "Automatic-Module-Name" to manifests of JAR files.
-* Add a Java 9 regression build.
 
 Tech Debt Reduction
 -------------------
 
-* Fix Travis build, remove Java 9 build, remove Sonar badge.
+* Deprecate or hide collection factory constructors.
```
Comment (nikhilnanivadekar) [inline RELEASE_NOTE_DRAFT.md]: Should this be under breaking change? I do not know, so asking.

Comment (donraab) [reply RELEASE_NOTE_DRAFT.md]: I assumed that was what your original comment meant - Line 42: This is a breaking change

Comment (donraab) [reply RELEASE_NOTE_DRAFT.md]: Ok fixing now.

**Pull request merged**